### PR TITLE
Isolate platform channels for macos

### PR DESCRIPTION
**Pull request by gaaclarke**
Adds integration tests for isolate platform channels on macos.

engine pr: https://github.com/flutter/engine/pull/35893

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by a-wallen** [comment]
(triage) looks like this one is still active and waiting for an engine PR to land.

**Comment by gaaclarke** [comment]
This is ready to review @a-wallen and @dnfield.  We just can't land it until the engine PR is merged.  The integration test isn't run as part of the presubmit tests.

**Review by dnfield — APPROVED** [review]

**Comment by gaaclarke** [comment]
The engine pr rolled in https://github.com/flutter/flutter/pull/112025

**Pull request merged**
